src/dist/matrix_ops.cpp

```diff
@@ -25,55 +23,89 @@ void rectangle_block(const Eigen::VectorXf &longDists,
                      const size_t nqqSamples, const size_t start,
                      const size_t max_elems);
 
-sparse_coo sparsify_dists(const NumpyMatrix &denseDists, const float distCutoff,
-                          const unsigned long int kNN) {
-  if (kNN > 0 && distCutoff > 0) {
-    throw std::runtime_error("Specify only one of kNN or distCutoff");
-  } else if (kNN < 1 && distCutoff < 0) {
-    throw std::runtime_error("kNN must be > 1 or distCutoff > 0");
+template <typename T>
+std::vector<T> combine_vectors(const std::vector<std::vector<T>> &vec,
+                               const size_t len) {
+  std::vector<T> all(len);
+  auto all_it = all.begin();
+  for (size_t i = 0; i < vec.size(); ++i) {
+    std::copy(vec[i].cbegin(), vec[i].cend(), all_it);
+    all_it += vec[i].size();
   }
+  return all;
+}
 
-  // ijv vectors
-  std::vector<float> dists;
-  std::vector<long> i_vec;
-  std::vector<long> j_vec;
-  if (distCutoff > 0) {
-    for (long i = 0; i < denseDists.rows(); i++) {
-      for (long j = i + 1; j < denseDists.cols(); j++) {
-        if (denseDists(i, j) < distCutoff) {
-          dists.push_back(denseDists(i, j));
-          i_vec.push_back(i);
-          j_vec.push_back(j);
-        }
-      }
+sparse_coo sparsify_dists(const NumpyMatrix &denseDists,
+                          const float distCutoff,
+                          const unsigned long int kNN,
+                          const size_t num_threads)
+{
+    if (kNN > 0 && distCutoff > 0)
+    {
+        throw std::runtime_error("Specify only one of kNN or distCutoff");
+    }
+    else if (kNN < 1 && distCutoff < 0)
+    {
+        throw std::runtime_error("kNN must be > 1 or distCutoff > 0");
```

Comment:
Brace formatting is inconsistent here. I usually use `clang-format` (you can add this into vscode to run on C++ files)

Reply:
Done